### DISCUSSION TITLE
Allow CreateNodes to be called just one time without file pattern

### DISCUSSION THREAD

**Discussion by klagrida** [nrwl/nx#23340] [Feature Requests]
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description

Allow `CreateNodes` to be called just one time without file pattern.

## Motivation

I found better in my case for java/kotlin workspace to run a command to get project dependencies before calling CreateNodes and CreateDependencies functions. Data are shared in a json file or cache between this two functions.

As a workaround I call `CreateNodes` with 'nx.json' pattern. But maybe other users are in the same case as me.

```
export const createNodes: CreateNodes = [
  'nx.json',
  () => {
....
```

**Comment by FrozenPandaz** [comment]
I've converted this to a discussion so we can discuss more easily.
